src/__tests__/Model.js

```diff
@@ -1733,6 +1737,121 @@ describe('requests', () => {
             expect(animal.hasUserChanges).toBe(true);
         });
     });
+
+    test('fetch handles out-of-order responses base', async () => {
+        const animal = new Animal(
+            { id: 1 },
+        );
+
+        // slow fetch, represents fetching an entire store
+        mock.onAny().replyOnce((res) => {
+            return delay(1000).then(() => {
+                return [200, { data: { id: 1, name: 'slow' } }];
+            });
+        });
+
+        // Request 2 - arrives immediately
+        mock.onAny().replyOnce(() => {
+            return [200, { data: { id: 1, name: 'fast' } }];
+        });
+
+        // Fire both requests roughly at the same time
+        const p1 = animal.fetch();
+        const p2 = animal.fetch();
+
+        await Promise.all([p1, p2]);
+
+        // p2 is called last, so will be set
+        expect(animal.name).toBe('fast');
+    });
+
+    test('fetch handles out-of-order responses pending request count', async () => {
+        const animal = new Animal(
+            { id: 1 },
+        );
+
+        // will never end, to increase request counter
+        mock.onAny().replyOnce((res) => {
+            return new Promise(() => {
+                // Never resolves
+            });
+        });
+
+        // Request 2 - arrives quickly
+        mock.onAny().replyOnce(() => {
+            return delay(1000).then(() => {
+                return [200, { data: { id: 1, name: 'fast' } }];
+            });
+        });
+
+        // Fire both requests roughly at the same time
+        const p1 = animal.fetch();
+        const p2 = animal.fetch();
+
+        expect(animal.__pendingRequestCount).toBe(2);
+        expect(animal.name).toBe('');
+
+        // p1 will never return
+        await p2
+
+        // check pending request counter
+        expect(animal.__pendingRequestCount).toBe(0);
+        expect(animal.name).toBe('fast');
+    });
+    test('fetch handles out-of-order responses pending request counter', async () => {
```

Comment:
This test doesn't check `__pendingRequestCount` at all. Can you fix the name?

Reply:
Hehe, accidental copy of another test it seems